pynucastro/rates/rate.py

```diff
@@ -1586,6 +1613,7 @@ def get_tabular_rate(self):
 
         # find .dat file and read it
         self.table_path = _find_rate_file(self.table_file)
+        self.source = RateSource.source(self.rfile_path.parent.name)
```

Comment:
This is the same as `RateSource.source(self.source)`, but will fail if `rfile` is an `io.StringIO`. Can it just be done where we set `self.source` in `__init__()`?